src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmBatchProcessorImpl.java

```diff
@@ -0,0 +1,316 @@
+
+        if (!allPendingSourceToArmEntities.isEmpty()) {
+
+            userAccount = userIdentity.getUserAccount();
+            File archiveRecordsFile = createEmptyArchiveRecordsFile();
+            var batchItems = new BatchItems();
+
+            for (var currentEod : allPendingSourceToArmEntities) {
+
+                var batchItem = new BatchItem();
+
+                try {
+                    ExternalObjectDirectoryEntity armEod;
+                    if (isEqual(currentEod.getExternalLocationType(), EodHelper.armLocation())) {
+                        armEod = updateArmEodToArmIngestionStatus(currentEod, batchItem, batchItems, archiveRecordsFile);
+                        if (armEod == null) {
+                            continue;
+                        }
+                    } else {
+                        armEod = createArmEodWithArmIngestionStatus(currentEod, batchItem, batchItems, archiveRecordsFile);
+                    }
+
+                    String rawFilename = generateFilename(armEod);
+                    if (shouldPushRawDataToArm(batchItem)) {
+                        pushRawDataAndCreateArchiveRecordIfSuccess(batchItem, rawFilename);
+                    } else if (shouldAddEntryToManifestFile(batchItem)) {
+                        var archiveRecord = archiveRecordService.generateArchiveRecordInfo(batchItem.getArmEod().getId(), rawFilename);
+                        batchItem.setArchiveRecord(archiveRecord);
+                    }
+                } catch (Exception e) {
+                    log.error("Unable to batch push EOD {} to ARM", currentEod.getId(), e);
+                    recoverByUpdatingEodToFailedArmStatus(batchItem);
+                }
+            }
+
+            try {
+                if (!batchItems.getSuccessful().isEmpty()) {
+                    writeManifestFile(batchItems, archiveRecordsFile);
+                    copyMetadataToArm(archiveRecordsFile);
+                }
+            } catch (Exception e) {
+                log.error("Error during generation of batch manifest file {}", archiveRecordsFile.getName(), e);
+                batchItems.getSuccessful().forEach(this::recoverByUpdatingEodToFailedArmStatus);
+                return;
+            }
+
+            for (var batchItem : batchItems.getSuccessful()) {
+                updateExternalObjectDirectoryStatus(batchItem.getArmEod(), EodHelper.armDropZoneStatus());
+            }
+        }
+
+        log.info("Finished running ARM Batch Push processing at: {}", OffsetDateTime.now());
+    }
+
+    private ExternalLocationTypeEntity getEodSourceLocation() {
+        var armClient = armDataManagementConfiguration.getArmClient();
+        if (armClient.equalsIgnoreCase("darts")) {
+            return EodHelper.unstructuredLocation();
+        } else if (armClient.equalsIgnoreCase("dets")) {
+            return EodHelper.detsLocation();
```

Comment:
This job is to move data from unstructured to ARM, so if the client is darts or dets they will both move to ARM? Also is there a possibility of a conflict between the 2?

Reply:
correct. The ticket requirements state that based on the value of the armClient config, the job needs to get the records either from unstructured or from dets. And push them to arm.
With regards to conflicts, I'll let @hemantasharma1129 reply. But I don't think